boards: xtensa: Add support for Audio DSP on Up Squared board

jhedberg opened the pull request:
Add support for the Intel Apollolake Audio DSP SoC and the Up Squared board
```
Andrei Emeltchenko (5):
      arch: xtensa: Use reset-vector.S in booloader code
      boards: up_squared_adsp: Add logging frontend
      log: Add Intel ADSP logging backend
      west: Add signing support using the rimage tool from SOF
      doc: up_squared_adsp: Add documentation for the board

Daniel Leung (1):
      dts: cpu: add binding for Cadence Tensilica Xtensa LX4 CPU

Johan Hedberg (4):
      CODEOWNERS: Add @dcpleung as owner for /dts/xtensa/intel/
      arch: xtensa: Add support for Intel Apollolake
      soc: xtensa: Add SoC definition for Audio DSP on Intel Apollolake
      boards: xtensa: Add board definition for Audio DSP on Up Squared

Tedd Ho-Jeong An (1):
      boards: up_squared_adsp: Add scripts for diagnostic driver
```

dcpleung committed:
dts: cpu: add binding for Cadence Tensilica Xtensa LX4 CPU

This adds a simple binding for the Cadence Tensilica Xtensa LX4
CPU. File originally from the LX6 binding.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

Commit:
arch: xtensa: Add support for Intel Apollolake

Add the necessary architecture changes for Intel Apollolake.

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

Commit:
soc: xtensa: Add SoC definition for Audio DSP on Intel Apollolake

Create an SoC definition for the Audio DSP on Intel Apollolake

Signed-off-by: Daniel Leung <daniel.leung@intel.com>
Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>
Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

finikorg committed:
log: Add Intel ADSP logging backend

Add logging backend for Intel Audio DSP.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
west: Add signing support using the rimage tool from SOF

This way "west sign" can be used for signing images.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
doc: up_squared_adsp: Add documentation for the board

Add documentation for the board up_squared_adsp.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

Commit:
boards: xtensa: Add board definition for Audio DSP on Up Squared

Create a board definition for the Audio DSP found on the Up Squared
board.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>
Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>
Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

finikorg committed:
boards: up_squared_adsp: Add logging frontend

Add a command-line frontend to match the logging backend for Intel
Apollolake-based boards like the Audio DSP in the Up Squared.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

tedd-an committed:
boards: up_squared_adsp: Add scripts for diagnostic driver

This patch adds scripts which downloads the firmware and extracts the
trace messages. The scripts required the SOF diagnostics driver
installed.

Signed-off-by: Tedd Ho-Jeong An <tedd.an@intel.com>

dcpleung commented:
Please remove these two:
- cmake: use x86_64 toolchain to build 32-bit targets
- cmake: xtensa: use SoC name in path to toolchain

As long as we are using SDK v0.11, this should build. There is no need for xtools at the moment.

jhedberg commented:
> Please remove these two:
> - cmake: use x86_64 toolchain to build 32-bit targets
> - cmake: xtensa: use SoC name in path to toolchain

@dcpleung what's the plan for those then? You'll be submitting them as an independent PR, or they can be completely forgotten about?

dcpleung commented:
> > Please remove these two:
> > 
> > * cmake: use x86_64 toolchain to build 32-bit targets
> > * cmake: xtensa: use SoC name in path to toolchain
> 
> @dcpleung what's the plan for those then? You'll be submitting them as an independent PR, or they can be completely forgotten about?

There are already PR @ https://github.com/zephyrproject-rtos/zephyr/pull/20807

jhedberg commented:
> There are already PR @ #20807

@dcpleung thanks for clarifying. The commits are now removed.

@finikorg @dcpleung please take a look at the remaining CI issues. The undefined CONFIG_* macros in the bootloader code are from SOF, I suppose. I'd just go and remove them (and any code behind them) but CONFIG_APOLLOLAKE has me a bit worried - can we really just remove that code, i.e. we don't need it?

@tedd-an there are still a few pylint issues for lib/device.py.

What do you guys think of the stdint type usage, which is the majority of the code warnings from CI? Should we just convert those to Zephyr types or there's a good reason to keep them?

dcpleung commented:
@jhedberg We need the code within CONFIG_APOLLOLAKE. You can just remove the #ifdef but keep the code.

jhedberg commented:
> @jhedberg We need the code within CONFIG_APOLLOLAKE. You can just remove the #ifdef but keep the code.

@dcpleung ok, though as anyone actually tested this code, since so far it's been compiled out? Was it being enabled in the combined build with SOF?

dcpleung commented:
> > @jhedberg We need the code within CONFIG_APOLLOLAKE. You can just remove the #ifdef but keep the code.
> 
> @dcpleung ok, though as anyone actually tested this code, since so far it's been compiled out? Was it being enabled in the combined build with SOF?

This is defined in the CMakeList.txt under bootloader.

jhedberg commented:
> This is defined in the CMakeList.txt under bootloader.

Ah, I didn't realise we still had such a hack in place. I'll remove that as well as do the necessary changes to the bootloader code.

finikorg commented:
@jhedberg For the shippable fix you probably need patches:
https://github.com/zephyrproject-rtos/zephyr/pull/21262/commits/d2dc7482294a8c934dde620202dbc1dfc59cc2b0
https://github.com/thesofproject/zephyr/pull/35/commits/22b20f87f0eff6cfd553f258e772a9802174d53d

jhedberg commented:
> This is defined in the CMakeList.txt under bootloader.

@dcpleung CMakeList.txt has `-DCONFIG_BUILD_VM_ROM` but this define is not used anywhere in the bootloader code. Instead, the code does have references to `CONFIG_VM_ROM`. Were those supposed to be the same? In that case we haven't been building what we thought we were building.

jhedberg commented:
```                                                                
if(CONFIG_XTENSA_HAL)
  target_link_libraries(bootloader PRIVATE XTENSA_HAL)
endif()
```

@dcpleung what about this branch in CMakeList.txt? CI complains of CONFIG_XTENSA_HAL since it's not defined in Kconfig? Should I remove these three lines from CMakeList.txt?

jhedberg commented:
@dcpleung just discovered that those three lines come from here:
https://github.com/thesofproject/zephyr/pull/31/commits/0f329ae65732c3f832842cde5aea139d933f5691

However, we obviously can't have them unless this CONFIG_* variable is defined somewhere in Kconfig.

finikorg commented:
@jhedberg the last commit https://github.com/zephyrproject-rtos/zephyr/pull/21241/commits/47b6fa4a4db6845e9f1127b93f70af120ac4409a is not needed.
See: https://github.com/zephyrproject-rtos/zephyr/pull/21262/commits/d2dc7482294a8c934dde620202dbc1dfc59cc2b0

dcpleung commented:
> > This is defined in the CMakeList.txt under bootloader.
> 
> @dcpleung CMakeList.txt has `-DCONFIG_BUILD_VM_ROM` but this define is not used anywhere in the bootloader code. Instead, the code does have references to `CONFIG_VM_ROM`. Were those supposed to be the same? In that case we haven't been building what we thought we were building.

Maybe you can trace SOF code and see how it is being used?

> ```
> if(CONFIG_XTENSA_HAL)
>   target_link_libraries(bootloader PRIVATE XTENSA_HAL)
> endif()
> ```
> 
> @dcpleung what about this branch in CMakeList.txt? CI complains of CONFIG_XTENSA_HAL since it's not defined in Kconfig. Should I remove these three lines from CMakeList.txt?

#21285 has `CONFIG_XTENSA_HAL` so maybe leave it here for now.

nordic-krch commented on subsys/logging/log_backend_adsp.c:
```diff
@@ -0,0 +1,154 @@
+static struct ring_buf ringbuf;
+
+static u16_t magic = 0x55aa;
+static u16_t log_id;
```
use only in trace(). Could be moved there.

nordic-krch commented on subsys/logging/log_backend_adsp.c:
```diff
@@ -0,0 +1,154 @@
+
+static void init(void)
+{
+	ring_buf_init(&ringbuf, CONFIG_LOG_BACKEND_ADSP_RINGBUF_SIZE,
```
should be somehow ensured that `CONFIG_LOG_BACKEND_ADSP_RINGBUF_SIZE` is a multiple of `BUF_SIZE` otherwise when buffer is claimed it might not have `BUF_SIZE` available even when ring buffer has `BUF_SIZE` available.

nordic-krch reviewed (commented):
Looked only at log backend. Looks ok with some minor comments

Commit:
CODEOWNERS: Add @dcpleung as owner for /dts/xtensa/intel/

Add Daniel Leung as code owner for all Intel xtensa dts files.

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

finikorg commented:
Something is broken in BSP since sanitycheck fails in this PR and works fine here: https://github.com/thesofproject/zephyr/pull/37

finikorg commented:
> Something is broken in BSP since sanitycheck fails in this PR and works fine here: [thesofproject#37](https://github.com/thesofproject/zephyr/pull/37)

I checked again and sanitycheck running with:
``
$ sanitycheck -p up_squared_adsp
``
with this PR it produces some errors:
``
INFO    - Total complete:  160/ 160  100%  skipped:   39, failed:  121
INFO    - 0 of 160 tests passed (0.00%), 121 failed, 39 skipped with 0 warnings in 99.64 seconds
INFO    - In total 738 test cases were executed on 1 out of total 217 platforms (0.46%)
``
So it seems there is a regression compared to the previous base my PR are based, they are all passing sanitycheck. Could we roll back to the last working version?

jhedberg commented:
> Could we roll back to the last working version?

@finikorg would that help with the upstreaming though, since this will ultimately anyway be rebased on top of master? I.e. wouldn't it be better to bisect, fix the issue and move forward rather than backward.

jhedberg commented:
```
INFO    - Total complete:  160/ 160  100%  skipped:   39, failed:    0
INFO    - 121 of 160 tests passed (100.00%), 0 failed, 39 skipped with 0 warnings in 3538.22 seconds
INFO    - In total 740 test cases were executed on 1 out of total 218 platforms (0.46%)
```

@finikorg there seems to be something different between your and my environments, since sanitycheck passes for me. Note that I also added a temporary patch to this PR to enable the 0.11-alpha-8 SDK version in Shippable, and the only failures are unrelated boards which apparently still have some bugs with the new SDK.

finikorg reviewed (approved):
Sorry, now we use `modules/hal/xtensa`, I had to use west update. Works now.

jhedberg commented:
@lgirdwood FYI

lgirdwood commented on boards/xtensa/up_squared_adsp/bootloader/boot_ldr.x:
```diff
@@ -0,0 +1,208 @@
+OUTPUT_ARCH(xtensa)
+MEMORY
```
I guess this does not need to be pre-processed for Zephyr ?

jhedberg replied:
Probably not - I think it was taken from SOF largely as-is. @finikorg any comment on this?

finikorg replied:
We have "bootloader" code, similar to SOF's one. That script places objects to the right places.

lgirdwood replied:
ok, FWIW SOF pre-processes linker scripts with some headers to align the sections etc. I'm good if you guys are happy.

andyross reviewed (approved):
Can't see anything wrong at the xtensa level, though I tend not to be a good reference on the platform boilerplate stuff.

mbolivar reviewed (approved):
sign.py changes LGTM. I won't review the big Python scripts in the boards/ directory though ;).

finikorg committed:
arch: xtensa: Use reset-vector.S in booloader code

Use BOOTLOADER definition to separate bootloader code. This allows to
use the same file reset-vector.S when building bootloader and when
CONFIG_XTENSA_RESET_VECTOR is enabled.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

jhedberg commented:
After a rebase to latest master Shippable fails with:
```
/opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/bin/ld: /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/lib/libc.a(lib_a-stdio.o):(.literal.__sread+0x4): undefined reference to `_read_r'
/opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/bin/ld: /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/lib/libc.a(lib_a-stdio.o):(.literal.__swrite+0x0): undefined reference to `_lseek_r'
/opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/bin/ld: /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/lib/libc.a(lib_a-stdio.o):(.literal.__swrite+0x4): undefined reference to `_write_r'
/opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/bin/ld: /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/lib/libc.a(lib_a-stdio.o):(.literal.__sclose+0x0): undefined reference to `_close_r'
/opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/bin/ld: /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/lib/libc.a(lib_a-stdio.o): in function `__sread':
stdio.c:(.text.__sread+0x1f): undefined reference to `_read_r'
/opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/bin/ld: /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/lib/libc.a(lib_a-stdio.o): in function `__swrite':
stdio.c:(.text.__swrite+0x32): undefined reference to `_lseek_r'
/opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/bin/ld: stdio.c:(.text.__swrite+0x5a): undefined reference to `_write_r'
/opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/bin/ld: /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/lib/libc.a(lib_a-stdio.o): in function `__sseek':
stdio.c:(.text.__sseek+0x1e): undefined reference to `_lseek_r'
/opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/bin/ld: /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/lib/libc.a(lib_a-stdio.o): in function `__sclose':
stdio.c:(.text.__sclose+0x1a): undefined reference to `_close_r'
collect2: error: ld returned 1 exit status
```

@galak it seems like this is likely because of your commit 9897c3b0dda296704cd1a8313683f292544c6e3c. Any idea what's going on and how to fix it?

finikorg commented:
> After a rebase to latest master Shippable fails with:
> 
> ```
> /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/bin/ld: /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/lib/libc.a(lib_a-stdio.o):(.literal.__sread+0x4): undefined reference to `_read_r'
> /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/bin/ld: /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/lib/libc.a(lib_a-stdio.o):(.literal.__swrite+0x0): undefined reference to `_lseek_r'
> /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/bin/ld: /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/lib/libc.a(lib_a-stdio.o):(.literal.__swrite+0x4): undefined reference to `_write_r'
> /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/bin/ld: /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/lib/libc.a(lib_a-stdio.o):(.literal.__sclose+0x0): undefined reference to `_close_r'
> /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/bin/ld: /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/lib/libc.a(lib_a-stdio.o): in function `__sread':
> stdio.c:(.text.__sread+0x1f): undefined reference to `_read_r'
> /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/bin/ld: /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/lib/libc.a(lib_a-stdio.o): in function `__swrite':
> stdio.c:(.text.__swrite+0x32): undefined reference to `_lseek_r'
> /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/bin/ld: stdio.c:(.text.__swrite+0x5a): undefined reference to `_write_r'
> /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/bin/ld: /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/lib/libc.a(lib_a-stdio.o): in function `__sseek':
> stdio.c:(.text.__sseek+0x1e): undefined reference to `_lseek_r'
> /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/bin/ld: /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/lib/libc.a(lib_a-stdio.o): in function `__sclose':
> stdio.c:(.text.__sclose+0x1a): undefined reference to `_close_r'
> collect2: error: ld returned 1 exit status
> ```

@nashif There are also some failures when building for `intel_s1000_crb` with:
```
$ sanitycheck -p intel_s1000_crb
```

galak commented:
Please fix this warning:

```
/home/galak/git/zephyr/include/generated_dts_board.h:1:2: warning: #warning "generated_dts_board.h has been renamed to devicetree.h. Please include tha
t instead." [-Wcpp]
    1 | #warning "generated_dts_board.h has been renamed to devicetree.h. Please include that instead."
      |  ^~~~~~~
```

galak commented:
> After a rebase to latest master Shippable fails with:
> 
> ```
> /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/bin/ld: /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/lib/libc.a(lib_a-stdio.o):(.literal.__sread+0x4): undefined reference to `_read_r'
> /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/bin/ld: /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/lib/libc.a(lib_a-stdio.o):(.literal.__swrite+0x0): undefined reference to `_lseek_r'
> /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/bin/ld: /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/lib/libc.a(lib_a-stdio.o):(.literal.__swrite+0x4): undefined reference to `_write_r'
> /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/bin/ld: /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/lib/libc.a(lib_a-stdio.o):(.literal.__sclose+0x0): undefined reference to `_close_r'
> /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/bin/ld: /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/lib/libc.a(lib_a-stdio.o): in function `__sread':
> stdio.c:(.text.__sread+0x1f): undefined reference to `_read_r'
> /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/bin/ld: /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/lib/libc.a(lib_a-stdio.o): in function `__swrite':
> stdio.c:(.text.__swrite+0x32): undefined reference to `_lseek_r'
> /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/bin/ld: stdio.c:(.text.__swrite+0x5a): undefined reference to `_write_r'
> /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/bin/ld: /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/lib/libc.a(lib_a-stdio.o): in function `__sseek':
> stdio.c:(.text.__sseek+0x1e): undefined reference to `_lseek_r'
> /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/bin/ld: /opt/sdk/zephyr-sdk-0.11.1/xtensa/intel_apl_adsp/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/9.2.0/../../../../xtensa-zephyr-elf/lib/libc.a(lib_a-stdio.o): in function `__sclose':
> stdio.c:(.text.__sclose+0x1a): undefined reference to `_close_r'
> collect2: error: ld returned 1 exit status
> ```
> 
> @galak it seems like this is likely related to your commit [9897c3b](https://github.com/zephyrproject-rtos/zephyr/commit/9897c3b0dda296704cd1a8313683f292544c6e3c). Any idea what's going on and how to fix it?

Weird that we don't see this on S1000, but here's a fix for the issue:

PR #22469

jhedberg commented:
@galak wrote:
> Please fix this warning:

Fixed. I also cherry-picked your build fix to the beginning of this PR.
